Property 'containers' does not exist on type 'PeaksOptions'

Hello, there is an error in my project when I execute ***pnpm build**, and this project made by ts + vite,

```
> vite build

vite v4.4.9 building for production...
transforming (423) node_modules\.pnpm\element-plus@2.3.12_vue@3.3.4\node_modules\element-plus\es\components\menu\src\use-menu.mjsnode_modules/vue-peaks/src/components/AudioPeaks.vue:130:20 - error TS2339: Property 'containers' does not exist on type 'PeaksOptions'.
  Property 'containers' does not exist on type 'OptionalOptions & RemoteWaveformDataOptions & Without<LocalWaveformDataOptions> & Without<WebAudioOptions> & ContainerOptions'.

130     (props.options.containers = {
                       ~~~~~~~~~~

node_modules/vue-peaks/src/components/AudioPeaks.vue:140:7 - error TS2322: Type 'PeaksOptions | { containers: { overview: HTMLDivElement | undefined; zoomview: HTMLDivElement | undefined; }; mediaElement: HTMLMediaElement | undefined; webAudio: { ...; }; zoomLevels: number[]; }' is not assignable to type 'PeaksOptions'.
  Type '{ containers: { overview: HTMLDivElement | undefined; zoomview: HTMLDivElement | undefined; }; mediaElement: HTMLMediaElement | undefined; webAudio: { ...; }; zoomLevels: number[]; }' is not assignable to type 'PeaksOptions'.
    Object literal may only specify known properties, and 'containers' does not exist in type 'OptionalOptions & Without<RemoteWaveformDataOptions> & Without<LocalWaveformDataOptions> & WebAudioOptions & ContainerOptions'.

140       containers: {
          ~~~~~~~~~~


Found 2 errors in the same file, starting at: node_modules/vue-peaks/src/components/AudioPeaks.vue:130

 ELIFECYCLE  Command failed with exit code 2.
transforming (523) node_modules\.pnpm\@floating-ui+dom@1.5.1\node_modules\@floating-ui\dom\dist\floating-ui.dom.mjsERROR: "type-check" exited with 1.
 ELIFECYCLE  Command failed with exit code 1.
```

My vue code:
```vue
    <audio ref="audioRef" :src="audioUrl" @canplay="onCanplay" id='externalMediaElement'></audio>

    <AudioPeaks mediaElementId='externalMediaElement' ></AudioPeaks>
```

Is there a wrong type defined in PeaksOptions?

--- a/README.md
+++ b/README.md
@@ -8,11 +8,11 @@
 [![](https://data.jsdelivr.com/v1/package/npm/vue-peaks/badge 'jsDelivr')](https://www.jsdelivr.com/package/npm/vue-peaks)
 [![Rate this package](https://badges.openbase.com/js/rating/vue-peaks.svg?token=vHkEYi5zzp1G84PyPGIiYYDN/9+SZtzXDlLAEe5ffRA=)](https://openbase.com/js/vue-peaks?utm_source=embedded&utm_medium=badge&utm_campaign=rate-badge)
 
-A lightweight Vue wrapper around the [bbc/peaks.js](https://github.com/bbc/peaks.js) audio waveform UI component.
+A lightweight Vue wrapper around the [bbc/peaks.js (v3)](https://github.com/bbc/peaks.js) audio waveform UI component.
 
 ![Image](https://github.com/suterma/vue-peaks/blob/main/vue-peaks-example-ui.png)
 
-vue-peaks aims to simplify the use of [peaks.js](https://github.com/bbc/peaks.js/) in a [Vue3]( https://vuejs.org/) environment. It provides a [Single File Component (SFC)](https://vuejs.org/guide/scaling-up/sfc.html), AudioPeaks.
+vue-peaks aims to simplify the use of [peaks.js](https://github.com/bbc/peaks.js/) in a [Vue3](https://vuejs.org/) environment. It provides a [Single File Component (SFC)](https://vuejs.org/guide/scaling-up/sfc.html), AudioPeaks.
 
 Explore the [live examples](https://suterma.github.io/vue-peaks/).
 
@@ -30,11 +30,12 @@ Installing vue-peaks also installs [peaks.js](https://github.com/bbc/peaks.js/) 
 # How to use
 
 ## Basic examples
+
 vue-peaks provides three modes of operation:
 
-*   **Simple mode**: just set the media URL, and all views are automatically rendered by default
-*   **Slot mode**: the media element and additional views are provided using named slots
-*   **External mode**: the media element and additional views are referenced by id or HTML element references
+- **Simple mode**: just set the media URL, and all views are automatically rendered by default
+- **Slot mode**: the media element and additional views are provided using named slots
+- **External mode**: the media element and additional views are referenced by id or HTML element references
 
 #### Just using a media source URL ("simple" mode)
 
@@ -115,28 +116,28 @@ npm run lint
 
 # Credits
 
-*   [peaks.js](https://github.com/bbc/peaks.js/), a client-side JavaScript component to display and interact with audio waveforms in the browser. (Released under the LGPL-3.0 license, by the British Broadcasting Corporation and contributors)
-*   [Vue.js](https://vuejs.org/), the progressive JavaScript framework. (Released under the MIT License, Copyright © 2014-2021 Evan You)
-*   [Bulma](https://bulma.io/), the modern CSS framework that just works. (Released under the MIT License, by Jeremy Thomas)
-*   [Bulmaswatch](https://jenil.github.io/bulmaswatch/), free themes for Bulma. (Released under the MIT License, by Jenil Gogari)
-*   [Material Design Icons](https://materialdesignicons.com/), the original. Following Google's Material Design guidelines for system icons. (Icons released under the Apache 2.0 License, by pictogrammers.com)
-*   [Jest](https://jestjs.io/), a delightful JavaScript Testing Framework. (Released under the MIT License, by Facebook)
-*   Music from the Album "Not For Sale" by Lidija Roos (Released to the public, by Lidija Roos)
-*   [@highlightjs/vue-plugin](https://github.com/highlightjs/vue-plugin), provides a highlightjs component for use in your Vue.js 3 applications. (Released under the BSD-3-Clause license, by the highlight.js team)
-*   [@rushstack/eslint-patch](https://github.com/microsoft/rushstack/tree/main/eslint/eslint-patch), a patch that improves how ESLint loads plugins when working in a monorepo with a reusable toolchain. (Released under the MIT License, by Microsoft)
-*   [@types/jsdom, @types/node](https://github.com/DefinitelyTyped/DefinitelyTyped), the repository for high quality TypeScript type definitions. (Released under the MIT License, by various contributors)
-*   [vite, vitest, @vitejs/plugin-vue](https://github.com/vitejs/vite), next Generation Frontend Tooling. (Released under the MIT License, by the vite team)
-*   [eslint](https://github.com/eslint/eslint), a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code. (Released under the MIT License, by the eslint team)
-*   [jsdom](https://github.com/jsdom/jsdom), a pure-JavaScript implementation of many web standards. (Released under the MIT License, by the jsdom team)
-*   [npm-run-all](https://github.com/mysticatea/npm-run-all), a CLI tool to run multiple npm-scripts in parallel or sequential. (Released under the MIT License, by Toru Nagashima)
-*   [prettier](https://github.com/prettier/prettier), an opinionated Code Formatter. (Released under the MIT License, by the prettier team)
-*   [typescript](https://github.com/Microsoft/TypeScript), a language for application-scale JavaScript. (Released under the Apache-2.0 license, by Microsoft)
-*   [vue-tsc](https://github.com/johnsoncodehk/volar), high-performance tooling for Vue. (Released under the MIT License, by Johnson Chu)
+- [peaks.js](https://github.com/bbc/peaks.js/), a client-side JavaScript component to display and interact with audio waveforms in the browser. (Released under the LGPL-3.0 license, by the British Broadcasting Corporation and contributors)
+- [Vue.js](https://vuejs.org/), the progressive JavaScript framework. (Released under the MIT License, Copyright © 2014-2021 Evan You)
+- [Bulma](https://bulma.io/), the modern CSS framework that just works. (Released under the MIT License, by Jeremy Thomas)
+- [Bulmaswatch](https://jenil.github.io/bulmaswatch/), free themes for Bulma. (Released under the MIT License, by Jenil Gogari)
+- [Material Design Icons](https://materialdesignicons.com/), the original. Following Google's Material Design guidelines for system icons. (Icons released under the Apache 2.0 License, by pictogrammers.com)
+- [Jest](https://jestjs.io/), a delightful JavaScript Testing Framework. (Released under the MIT License, by Facebook)
+- Music from the Album "Not For Sale" by Lidija Roos (Released to the public, by Lidija Roos)
+- [@highlightjs/vue-plugin](https://github.com/highlightjs/vue-plugin), provides a highlightjs component for use in your Vue.js 3 applications. (Released under the BSD-3-Clause license, by the highlight.js team)
+- [@rushstack/eslint-patch](https://github.com/microsoft/rushstack/tree/main/eslint/eslint-patch), a patch that improves how ESLint loads plugins when working in a monorepo with a reusable toolchain. (Released under the MIT License, by Microsoft)
+- [@types/jsdom, @types/node](https://github.com/DefinitelyTyped/DefinitelyTyped), the repository for high quality TypeScript type definitions. (Released under the MIT License, by various contributors)
+- [vite, vitest, @vitejs/plugin-vue](https://github.com/vitejs/vite), next Generation Frontend Tooling. (Released under the MIT License, by the vite team)
+- [eslint](https://github.com/eslint/eslint), a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code. (Released under the MIT License, by the eslint team)
+- [jsdom](https://github.com/jsdom/jsdom), a pure-JavaScript implementation of many web standards. (Released under the MIT License, by the jsdom team)
+- [npm-run-all](https://github.com/mysticatea/npm-run-all), a CLI tool to run multiple npm-scripts in parallel or sequential. (Released under the MIT License, by Toru Nagashima)
+- [prettier](https://github.com/prettier/prettier), an opinionated Code Formatter. (Released under the MIT License, by the prettier team)
+- [typescript](https://github.com/Microsoft/TypeScript), a language for application-scale JavaScript. (Released under the Apache-2.0 license, by Microsoft)
+- [vue-tsc](https://github.com/johnsoncodehk/volar), high-performance tooling for Vue. (Released under the MIT License, by Johnson Chu)
 
 Initial code is taken from
 
--   [peaks-vue-demo](https://github.com/candeogi/peaks-vue-demo) by Giovanni Candeo (MIT-licensed)
--   [vue-waveform-template](https://github.com/thom4parisot/vue-waveform-template) by Thomas Parisot (MIT-licensed)
+- [peaks-vue-demo](https://github.com/candeogi/peaks-vue-demo) by Giovanni Candeo (MIT-licensed)
+- [vue-waveform-template](https://github.com/thom4parisot/vue-waveform-template) by Thomas Parisot (MIT-licensed)
 
 ---
 

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "vue-peaks",
-  "version": "1.0.2",
+  "version": "2.0.0",
   "description": "A lightweight Vue wrapper around the bbc/peaks.js audio waveform UI component",
   "repository": {
     "type": "git",
@@ -31,7 +31,7 @@
     "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore"
   },
   "dependencies": {
-    "peaks.js": "^2.0.6",
+    "peaks.js": "^3.0.0",
     "vue": "^3.2.39"
   },
   "devDependencies": {

--- a/src/components/AudioPeaks.vue
+++ b/src/components/AudioPeaks.vue
@@ -83,18 +83,17 @@ onMounted(() => {
   createPeaksInstance();
 });
 
-
 onBeforeUnmount(() => {
   destroyPeaksInstance();
 });
 
 const emit = defineEmits<{
   /**
- * Triggers in case the initialization fails
- *
- * @property {Error} error The error
- */
-  (e: 'error', error: Error): void
+   * Triggers in case the initialization fails
+   *
+   * @property {Error} error The error
+   */
+  (e: 'error', error: Error): void;
 }>();
 
 /** Initializes the peaks instance
@@ -127,35 +126,38 @@ function createPeaksInstance(): void {
   );
 
   if (props.options) {
-    (props.options.containers = {
-      overview: overviewElement,
-      zoomview: zoomviewElement,
-    }),
-      (props.options.mediaElement = mediaElement);
+    props.options.zoomview = {
+      container: zoomviewElement,
+    };
+    props.options.overview = {
+      container: overviewElement,
+    };
+    props.options.mediaElement = mediaElement;
   }
 
   const options: PeaksOptions = props.options
     ? props.options
     : {
-      containers: {
-        overview: overviewElement,
-        zoomview: zoomviewElement,
-      },
-      mediaElement: mediaElement,
-      webAudio: {
-        audioContext: new AudioContext(),
-      },
-      zoomLevels: [256, 512, 1024, 2048, 4096],
-    };
+        overview: { container: overviewElement },
+        zoomview: { container: zoomviewElement },
+        mediaElement: mediaElement,
+        webAudio: {
+          audioContext: new AudioContext(),
+        },
+        zoomLevels: [256, 512, 1024, 2048, 4096],
+      };
 
-  Peaks.init(options, function (err: Error, peaks: PeaksInstance | undefined): void {
-    if (err) {
-      console.error(err);
-      emit('error', err)
+  Peaks.init(
+    options,
+    function (err: Error, peaks: PeaksInstance | undefined): void {
+      if (err) {
+        console.error(err);
+        emit('error', err);
+      }
+      peaksInstance.value = peaks;
+      zoomLevel.value = peaks?.zoom.getZoom();
     }
-    peaksInstance.value = peaks;
-    zoomLevel.value = peaks?.zoom.getZoom();
-  });
+  );
 }
 
 /** Destroys the peaks instance
@@ -233,23 +235,42 @@ function zoomOut(): void {
   <div>
     <div ref="overviewSlot">
       <!-- @slot Named slot for the overview element. If an external overview element is referenced, the overview slot is not used -->
-      <slot name="overview" v-if="!props.overviewElementId && !props.overviewElement">
-        <div class="peaks-overview" ref="overview"></div>
+      <slot
+        name="overview"
+        v-if="!props.overviewElementId && !props.overviewElement"
+      >
+        <div
+          class="peaks-overview"
+          ref="overview"
+        ></div>
       </slot>
     </div>
 
     <div ref="zoomviewSlot">
       <!-- @slot Named slot for the zoomview element. If an external zoomview element is referenced, the zoomview slot is not used -->
-      <slot name="zoomview" v-if="!props.zoomviewElementId && !props.zoomviewElement">
-        <div class="peaks-zoomview" ref="zoomview"></div>
+      <slot
+        name="zoomview"
+        v-if="!props.zoomviewElementId && !props.zoomviewElement"
+      >
+        <div
+          class="peaks-zoomview"
+          ref="zoomview"
+        ></div>
       </slot>
     </div>
 
     <div ref="audioSlot">
       <!-- @slot Default slot for the media element. If an external media element is referenced, the default slot is not used -->
-      <slot name="default" v-if="!props.mediaElementId && !props.mediaElement">
+      <slot
+        name="default"
+        v-if="!props.mediaElementId && !props.mediaElement"
+      >
         <!-- The default content slot for the "slot" mode -->
-        <audio class="peaks-audio" ref="audio" controls>
+        <audio
+          class="peaks-audio"
+          ref="audio"
+          controls
+        >
           <source :src="src" />
         </audio>
       </slot>

